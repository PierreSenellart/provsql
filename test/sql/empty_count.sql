\set ECHO none
\pset format unaligned
SET search_path TO provsql_test, provsql;

-- From https://github.com/PierreSenellart/provsql/issues/60

CREATE TABLE test (id INT PRIMARY KEY);
SELECT add_provenance('test');
SELECT create_provenance_mapping('pm', 'test', 'id');

INSERT INTO test (id) VALUES (1), (2), (3), (4), (5);

CREATE TABLE test2 as select * FROM test WHERE id = 6;
SELECT id, sr_formula(provenance(), 'pm') FROM test2;

CREATE TABLE test3 as select count(*) FROM test WHERE id = 6;
CREATE TABLE result_test AS
SELECT count, sr_formula(provenance(), 'pm') FROM test3;
SELECT remove_provenance('result_test');
SELECT * FROM result_test;

DROP TABLE test;
DROP TABLE test2;
DROP TABLE test3;
DROP TABLE pm;
DROP TABLE result_test;
