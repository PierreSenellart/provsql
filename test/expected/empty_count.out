\set ECHO none
add_provenance

(1 row)
create_provenance_mapping

(1 row)
id|sr_formula|provsql
(0 rows)
remove_provenance

(1 row)
count|sr_formula
|𝟙
(1 row)
